src/vs/platform/actions/browser/buttonbar.ts

```diff
@@ -128,12 +135,14 @@ export class WorkbenchButtonBar extends ButtonBar {
 					btn.element.classList.add(...action.class.split(' '));
 				}
 			}
-			const kb = this._keybindingService.lookupKeybinding(action.id);
-			let tooltip: string;
-			if (kb) {
-				tooltip = localize('labelWithKeybinding', "{0} ({1})", action.tooltip || action.label, kb.getLabel());
-			} else {
-				tooltip = action.tooltip || action.label;
+
+			if (!tooltip) {
+				const kb = this._keybindingService.lookupKeybinding(action.id);
```

Comment:
Seems like you could avoid the duplicated code by doing this before the other stuff.